src/screens/time/components/EditPunchInOutDateTimeCardComponent.tsx

```diff
@@ -0,0 +1,303 @@
+                    styles.flexOne,
+                    {
+                      paddingLeft: theme.spacing * 3,
+                    },
+                  ]}>
+                  <View
+                    style={{
+                      padding: theme.spacing,
+                      paddingRight: theme.spacing * 3,
+                    }}>
+                    <Icon name={'clock-outline'} fontSize={20} />
+                  </View>
+                  <View style={{}}>
+                    <Text style={{fontSize: theme.spacing * 4.5}}>
+                      {'Time'}
+                    </Text>
+                  </View>
+                </View>
+                <View style={[styles.flexTwo, styles.rowFlexDirection]}>
+                  <View
+                    style={[
+                      styles.alignItemsFlexEnd,
+                      styles.flexOne,
+                      {
+                        paddingRight: theme.spacing,
+                      },
+                    ]}>
+                    <Text
+                      style={{
+                        color: theme.palette.secondary,
+                        fontSize: theme.spacing * 4.5,
+                      }}>
+                      {timeDisplay}
+                    </Text>
+                  </View>
+                  <View style={[styles.alignItemsFlexEnd]}>
+                    <Icon name={'chevron-right'} fontSize={theme.spacing * 5} />
+                  </View>
+                </View>
+              </TouchableOpacity>
+              {this.state.show ? (
+                <>
+                  <View>
+                    <DateTimePicker
+                      testID="dateTimePicker"
+                      value={date}
+                      mode={mode}
+                      is24Hour={false}
+                      display="default"
+                      onChange={this.onChange}
+                    />
+                  </View>
+                </>
+              ) : null}
+            </CardContent>
+          </Card>
+        </View>
+      </>
+    );
+  }
+}
+
+interface EditPunchInOutDateTimeCardProps extends WithTheme {
+  punchCurrentDateTime?: Date;
+  updateDateTime: (date: Date) => void;
+}
+
+interface EditPunchInOutDateTimeCardState {
+  show: boolean;
+  mode: typeof TIME | typeof DATE;
+}
+
+const styles = StyleSheet.create({
+  validPeriodText: {
+    alignItems: 'center',
+  },
+
+  justifyContentCenter: {
+    justifyContent: 'center',
+  },
+
+  alignItemsFlexEnd: {
+    alignItems: 'flex-end',
+  },
+
+  rowFlexDirection: {
+    flexDirection: 'row',
+  },
+  flexFour: {
+    flex: 4,
+  },
+  flexTwo: {
+    flex: 2,
+  },
+  flexThree: {
+    flex: 3,
+  },
+  flexOne: {
+    flex: 1,
+  },
+  flexFive: {
+    flex: 5,
+  },
+  flexSix: {
+    flex: 6,
+  },
+});
+
+const mapStateToProps = () => ({});
+
+const mapDispatchToProps = {
+  fetchPunchStatus,
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
```

Comment:
you can use `null` instead declaring `mapStateToProps`.
`const connector = connect(null, mapDispatchToProps);`